BioChem/package.mo

```diff
@@ -139,7 +122,7 @@ Version 1.0.1 (2013-04-18)
 <li>Version 0.96 (2008-12-19)</li>
 </ul>
  </body></html>"),
-    uses(Modelica(version = "3.2.3")),
+    uses(Modelica(version = "4.0.0")),
```

Comment:
This cannot be changed without also running the conversion script. Running the conversion script will also move annotations around (because they are not at the end of the class in most BioChem libraries). I could apply this conversion in a few separate PRs if you want.

Reply:
Not really sure what is meant by conversion script? What is it? and how to run it?

Reply:
See [Section 18.8.2](https://specification.modelica.org/maint/3.5/annotations.html#version-handling) of the Modelica Specification

Reply:
To my understanding now, the conversion script should convert: 

`annotation(uses(Modelica(version="3.2.3")));`

to 

`annotation(uses(Modelica(version="4.0.0"))); `

Is this the case, or should the script convert something else? 
Also I think this annotation exist only in one place, so why having a conversion script? 

Say we need to apply a conversion script, should I write one, or do we have a generic conversion script somewhere? 

Should Biochem.mo includes the following command: 

`conversion(from(version = "3.2.3", to= "4.0.0"", script = "ConvertMSL3.2.3To4.0.0.mos"))`

what should ConvertMSL3.2.3To4.0.0.mos  include?

Reply:
The conversion script makes changes to the library. For example, `Modelica.SIunits` changed name to `Modelica.Units.SI`. No conversion script is needed in most cases (if you extend from something that changed interface, a new conversion script might be needed).

Reply:
Biochem has its own physical units package. 

Is there a particular conversion script somewhere I need to apply?

Reply:
Yes, `modelica://Modelica/Resources/Scripts/Conversion/ConvertModelica_from_3.2.3_to_4.0.0.mos`. I can apply that in a separate pair of pull requests. The changes applied are quite extensive (~140 kB diff file). It's a lot of things like:
```diff
-  type Pressure= Modelica.SIunits.Pressure annotation(Icon(coordinateSystem(extent={{-100,100},{100,-100}}, preserveAspectRatio=true, grid={10,10})), Diagram(coordinateSystem(extent={{-100,100},{100,-100}}, preserveAspectRatio=true, grid={10,10})));
+  type Pressure= Modelica.Units.SI.Pressure annotation(Icon(coordinateSystem(extent={{-100,100},{100,-100}}, preserveAspectRatio=true, grid={10,10})), Diagram(coordinateSystem(extent={{-100,100},{100,-100}}, preserveAspectRatio=true, grid={10,10})));
```

Reply:
Have changed to MSL 3.2.3

Reply:
I have attempted to execute the conversion script. Not sure how to do it correctly, since the associated description file does not state how to run it. 

I have tried it with OMShell. Simply copying the script to the top folder (and again where package.mo is located): 

` >>> runScript("ConvertModelica_from_3.2.3_to_4.0.0.mos")`

 It does not work. 

My other attempt is to extend the annotation (in the same way as MSL) as follows: 

```
    uses(Modelica(version = "3.2.3")),
    version = "1.0.2",
    conversion(
 from(version = "3.2.3", script="modelica://Modelica/Resources/Scripts/Conversion/ConvertModelica_from_3.2.3_to_4.0.0.mos")),
    Icon( ... 
```

I get a statement in OMEdit that OpenModelica currently does not support conversion scripts 

How to execute the script?

Reply:
If you use the OpenModelica master, OMC can convert libraries in memory (but not save to file). I have some scripts locally for this.

Reply:
Recent commits have been integrated